components/calendar/src/types.rs

```diff
@@ -11,198 +11,215 @@ use tinystr::TinyAsciiStr;
+        /// the same month.
+        ///
+        /// Note: using [`Self::month_code`] is recommended, because the ordinal month numbers
+        /// can vary from year to year, as illustrated in the following example.
+        ///
+        /// # Examples
+        ///
+        /// Either `month_code` or `ordinal_month` can be used in DateFields, but they
+        /// might not resolve to the same month number:
+        ///
+        /// ```
+        /// use icu::calendar::Date;
+        /// use icu::calendar::cal::ChineseTraditional;
+        /// use icu::calendar::types::DateFields;
+        ///
+        /// // The 2023 Year of the Rabbit had a leap month after the 2nd month.
+        /// let mut fields1 = DateFields::default();
+        /// fields1.extended_year = Some(2023);
+        /// fields1.month_code = Some(b"M02L");
+        /// fields1.day = Some(1);
+        ///
+        /// let date1 = Date::try_from_fields(
+        ///     fields1,
+        ///     Default::default(),
+        ///     ChineseTraditional::new()
+        /// )
+        /// .expect("a well-defined Chinese date from month code");
+        ///
+        /// let mut fields2 = DateFields::default();
+        /// fields2.extended_year = Some(2023);
+        /// fields2.ordinal_month = Some(3);
+        /// fields2.day = Some(1);
+        ///
+        /// let date2 = Date::try_from_fields(
+        ///     fields2,
+        ///     Default::default(),
+        ///     ChineseTraditional::new()
+        /// )
+        /// .expect("a well-defined Chinese date from ordinal month");
+        ///
+        /// assert_eq!(date1, date2);
+        ///
+        /// let month_info = date1.month();
+        /// assert_eq!(month_info.ordinal, 3);
+        /// assert_eq!(month_info.standard_code.0, "M02L");
+        /// ```
+        pub ordinal_month: Option<u8>,
+        /// See [`DayOfMonth`].
+        pub day: Option<u8>,
+    }
 
-// Custom impl to stringify era and month_code where possible.
-impl fmt::Debug for DateFields<'_> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        // Ensures we catch future fields
-        let Self {
-            era,
-            era_year,
-            extended_year,
-            month_code,
-            ordinal_month,
-            day,
-        } = *self;
-        let mut builder = f.debug_struct("DateFields");
-        if let Some(s) = era.and_then(|s| core::str::from_utf8(s).ok()) {
-            builder.field("era", &Some(s));
-        } else {
-            builder.field("era", &era);
-        }
-        builder.field("era_year", &era_year);
-        builder.field("extended_year", &extended_year);
-        if let Some(s) = month_code.and_then(|s| core::str::from_utf8(s).ok()) {
-            builder.field("month_code", &Some(s));
-        } else {
-            builder.field("month_code", &month_code);
+    // Custom impl to stringify era and month_code where possible.
+    impl fmt::Debug for DateFields<'_> {
```

Comment:
Nit optional: keep the impls outside the module. Only put the definition inside the module since you are forced to